### PR TITLE
Rodrigom/benchmarking switching time

### PR DESCRIPTION
**Pull request by snova-rodrigom**
- Pulled from old repo
    - Added logic and adapted for switching time calculation
    - Added logic and adapted for running multiple experts under the same endpoint
    - Added code to analyze switching time in notebook running multiple models
- Adapted code with current main version
- Ran ruff and mypy

### CODE REVIEW

File: benchmarking/run_synthetic_dataset.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 python src/evaluator.py \
 --mode synthetic \
---model-name "llama3-405b" \
+--model-names "llama3-8b llama3-70b	llama3-405b" \
```
Comment (snova-amitk) [inline benchmarking/run_synthetic_dataset.sh:6]: Let's put the sambastudio model names by default

**Review by snova-amitk — CHANGES_REQUESTED** [review]
Some minor change around making sambastudio the default for model-names

File: benchmarking/run_synthetic_dataset.sh (hunk 1 above)
Comment (snova-rodrigom) [reply benchmarking/run_synthetic_dataset.sh:6]: thanks! done!

**Review by snova-amitk — APPROVED** [review]

**Review by snova-kwasia — APPROVED** [review]
LGTM @snova-rodrigom

**Pull request merged**